SpatialGDK/Extras/schema/component_presence.schema

```diff
@@ -1,13 +1,19 @@
 // Copyright (c) Improbable Worlds Ltd, All Rights Reserved
 package unreal;
 
-// The ComponentPresence component should be present on all entities that 
-// should be load-balanced. This contains a single property which is the 
-// list of component IDs that are present on the entity. Currently, this 
-// is used for enabling dynamic components in a multi-worker environment,
-// and should be useful in future for deducing entity completeness without
-// critical sections.
+// The ComponentPresence component should be present on all entities.
 component ComponentPresence {
     id = 9972;
+
+    // The component_list is a list of component IDs that should be present on
+    // this entity. This should be useful in future for deducing entity completeness
+    // without critical sections but is used currently just found enabling dynamic
+    // components in a multi-worker environment.
     list<uint32> component_list = 1;
+
+    // The possessing_client_worker_id is an optional worker ID string that is set 
+    // by the simulating worker when the Actor becomes net-owned by a client connection.
+    // The enforcer uses this value to update the EntityACL entry for the client RPC
+    // endpoint (and Heartbeat component, if present).
+    option<string> possessing_client_worker_id = 2;
```

Comment:
This really doesn't feel like it belongs in the ComponentPresence schema. I always think about that component as about what is needed to be able to simulate the entity. This is a little bit of that because of the RPC delivery, but it feels much more like it belongs in the AuthorityIntent.

Reply:
Although I guess it is an odd mismatch that the AuthorityIntent is used only for server workers, and this is used only for clients. hmm...

Reply:
I considered putting it in AuthorityIntent, I guess my consideration was that with USLB it'll be removed, and remaining things related to migration / mutli-server authority management like the component_presence list and possessing client worker ID it should sit together. Happy to move to AuthorityIntent now, or maybe we should just change the ComponentPresence component_list property to be component_presence and rename the component itself to something more generic?

Reply:
I don't have a super strong opinion about this, since there doesn't seem to be a perfect place for it. I would have a slight preference to rename the property/component as you suggested, but maybe run by Jacques to see his opinion on that one.

Reply:
Component presence and ownership aren't really the same thing and should be on different components IMO. They are written to and read by the same workers but they don't really share semantic meaning other than they communicate with the ACL worker.

Reply:
Similarly this shouldn't be on authority intent unless we want to put them all on the same component. But then, as mentioned we might want clients to read some of these. Although for entity completeness stuff the latest ideas we've been talking about don't actually rely on reading a presence component.

Reply:
Yea they're going into diff components